### PR TITLE
[no-release-notes] keyless merges and fixes for merge related tables in the new format

### PR DESCRIPTION
**Pull request by druvv**

### CODE REVIEW

File: integration-tests/bats/keyless-foreign-keys.bats [hunk 1]
```diff
@@ -1,1700 +1,1699 @@
...
-    dolt add .
-    dolt commit -m 'initial'
-
-    dolt sql -q 'insert into unprocessed_t values (20)'
-    dolt sql -q 'update t set additional_id = null, unprocessed_id = 20 where id = 1'
-    dolt sql -q 'delete from additional_t'
-    dolt add .
-    dolt commit -m 'this should not break'
-}
+#!/usr/bin/env bats
```
Comment (andrew-wm-arthur) [inline integration-tests/bats/keyless-foreign-keys.bats:1]: I can't figure out why this whole file is a diff...

File: go/store/prolly/tree/merge.go [hunk 2]
```diff
@@ -177,14 +177,12 @@ func sendPatches(ctx context.Context, l, r Differ, buf patchBuffer, cb Collision
...
 		case cmp == 0:
-			if !equalDiffVals(left, right) {
-				resolved, ok := cb(left, right)
-				if ok {
-					err = buf.sendPatch(ctx, resolved)
-				}
-				if err != nil {
-					return err
-				}
+			resolved, ok := cb(left, right)
```
Comment (andrew-wm-arthur) [inline go/store/prolly/tree/merge.go:180]: +1

**Review by andrew-wm-arthur — APPROVED** [review]
LGTM!

File: integration-tests/bats/keyless-foreign-keys.bats (hunk 1 above)
Comment (druvv) [reply integration-tests/bats/keyless-foreign-keys.bats:1]: the entire file had a space after each line, none of the other bats file do, so...

**Pull request merged**